jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ProxyConnectionFactory.java

```diff
@@ -643,32 +665,42 @@ private void parseBodyAndUpgrade() throws IOException
+                            uniqueId = StringUtil.toHexString(value);
+
+                        if (type == PP2_TYPE_SSL)
                         {
-                            int client = value[0] & 0xFF;
-                            if (client == ProxyEndPoint.PP2_TYPE_SSL_PP2_CLIENT_SSL)
+                            client = value[0] & 0xFF;
+                            sslVerified = value[1] == 0 && value[2] == 0 && value[3] == 0 && value[4] == 0;
+                            sslVersion = null;
+                            sslDistinguishedName = null;
+                            sslCipher = null;
+                            int i = 5; // Index of the first sub_tlv, after verify.
+                            while (i < length)
                             {
-                                int i = 5; // Index of the first sub_tlv, after verify.
-                                while (i < length)
+                                int subType = value[i++] & 0xFF;
+                                int subLength = (value[i++] & 0xFF) * 256 + (value[i++] & 0xFF);
+                                byte[] subValue = new byte[subLength];
+                                System.arraycopy(value, i, subValue, 0, subLength);
+                                i += subLength;
+                                switch (subType)
                                 {
-                                    int subType = value[i++] & 0xFF;
-                                    int subLength = (value[i++] & 0xFF) * 256 + (value[i++] & 0xFF);
-                                    byte[] subValue = new byte[subLength];
-                                    System.arraycopy(value, i, subValue, 0, subLength);
-                                    i += subLength;
-                                    if (subType == ProxyEndPoint.PP2_SUBTYPE_SSL_VERSION)
-                                    {
-                                        String tlsVersion = new String(subValue, StandardCharsets.US_ASCII);
-                                        proxyEndPoint.setAttribute(TLS_VERSION, tlsVersion);
-                                    }
+                                    case PP2_SUBTYPE_SSL_VERSION -> sslVersion = new String(subValue, StandardCharsets.US_ASCII);
+                                    case PP2_SUBTYPE_SSL_CN -> sslDistinguishedName = new String(subValue, StandardCharsets.UTF_8);
+                                    case PP2_SUBTYPE_SSL_CIPHER -> sslCipher = new String(subValue, StandardCharsets.US_ASCII);
                                 }
                             }
                         }
                     }
 
+                    // TODO should client certificate information be passed? For now we just check there is a client value
+                    //      and the certificate has been verified
+                    if (client != 0 && sslVerified)
+                        proxyEndPoint.setSecure(sslVersion, uniqueId, sslDistinguishedName, sslCipher);
```

Comment:
That is mostly what I've done, hence the TODO.

The changes I have made here have been to fix bugs.  Specifically we need to check the validate value, else we are saying a connection is secure when it has not been validated.

OK, I also added the handling of cipher name, because that is what this PR is about... common handling of SslSessionData.   The proxy protocol provides it, so we should use it.      Anything we don't have handling for I have TODO'd